File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        int minSize = int.MaxValue;
        bool found = false;
        int windowSize = 0;
        int windowSum = 0;
        int start = 0;
        
        for(int end = 0; end < nums.Length; end++){
            
            windowSum += nums[end];
            windowSize++;
            
            while(windowSum >= target && start <= end){
                found = true;
                minSize = Math.Min(windowSize, minSize);
                windowSum -= nums[start];
                start++;
                windowSize--;
            }
        }
        
        return found ? minSize : 0;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        minSize = sys.maxsize
        found = False
        windowSize = 0
        windowSum = 0
        start = 0
        
        for end in range(len(nums)):
            
            windowSum += nums[end]
            windowSize += 1
            
            while windowSum >= target:
                found = True
                minSize = min(windowSize, minSize)
                windowSum -= nums[start]
                start += 1
                windowSize -= 1
            
        return minSize if found else 0